### DISCUSSION TITLE
Share Your Creations 🎉

### DISCUSSION THREAD

**Discussion by XAMPPRocky** [EmbarkStudios/rust-gpu#517] [Show and tell]
This a thread to show off any creations you've made with `rust-gpu`! 🐉 Whether that's a shader written in Rust, or a project using `rust-gpu`.

For including screenshots you can control the size of the image by using an `img` HTML and setting the width to a percentage.

```html
<img alt="" src="" width="60%">
```

I'll start with including the previous picks for the release notes.

<img alt="v0.3 helmet" src="https://user-images.githubusercontent.com/4464295/111952909-256ddc00-8ae6-11eb-8774-55da6e523954.png" width="60%">

_The Khronos Sci-Fi helmet model. glTF model viewer created by @msiglreith using `rust-gpu` ([Source](https://github.com/msiglreith/grr-gltf))_

<img alt="v0.2 shader toys" src="https://user-images.githubusercontent.com/4464295/111952923-2999f980-8ae6-11eb-9210-f007f77f14df.png" width="60%">

_Created by @bcata6 using `rust-gpu` ([Rust source](https://github.com/LykenSol/rust-gpu-shadertoys)) with shaders originally from https://shadertoy.com/_

<img alt="v0.1 sky" src="https://user-images.githubusercontent.com/4464295/111952931-2d2d8080-8ae6-11eb-97fa-0ac2fba4a231.jpg" width="60%">

*Simple Sky shader created by @Jasper-Bekkers  ([Source](https://github.com/EmbarkStudios/rust-gpu/tree/main/examples/shaders/sky-shader))*

**Comment by Fahien** [comment]
Hi there!

I have been playing with rust-gpu recently on my hobby project: [VKR](https://github.com/Fahien/vkr), a Rust Vulkan renderer.
Nothing fancy, just simple 2D animations for the moment.

![vkr-v0 2 0](https://user-images.githubusercontent.com/6058008/114401400-2e147800-9ba3-11eb-80b0-6239aee1b851.gif)

https://user-images.githubusercontent.com/6058008/114401421-31a7ff00-9ba3-11eb-887f-e56412f45d96.mp4

**Comment by repi** [comment]
A few first pictures of the _creative platform_ we are building at Embark built fully in Rust including most of the shaders for the renderer! `rust-gpu` with Vulkan and hardware raytracing for real-time dynamic global illumination and reflections 🎉

More details about it in this blog post from @h3r2tic : https://medium.com/embarkstudios/homegrown-rendering-with-rust-1e39068e56a7

![story_1 with logo](https://user-images.githubusercontent.com/1262692/146794816-09afac39-f296-44c4-9966-f1b121411ab4.png)

![story_4 reflective with logo](https://user-images.githubusercontent.com/1262692/146794961-6afb5317-1ca0-42b3-8ced-cd06b6e945cb.png)

![seal_w_logo](https://user-images.githubusercontent.com/1262692/146808931-4867bf30-425e-4b81-8434-51a6127d8515.jpg)

![water2-day_w_logo](https://user-images.githubusercontent.com/1262692/146808969-f418905d-10af-43cf-a0e9-49f65f5cb128.png)

**Comment by SiebenCorgie** [comment]
I am using rust-gpu in my signed-distance-field library/renderer.
I serialize the signed distance function to a byte-code that is then interpreted on the GPU.
It is possible to define 2D SDFs, 3D SDFs and project 2D SDFs into 3D.

All of the shaders are currently written in rust and compiled using rust-gpu. They can be found [here](https://gitlab.com/tendsinmende/nako/-/tree/main/crates/nako_shader). Keep in mind that this a one-person hobby project. For some shader inspiration, have a look around!

I am currently experimenting with segment tracing, as well as better ways to upload and use the SDFs on the GPU.

Left to right: Simple sphere, a sphere used as intersection volume for a cross, the same cross, but using a smooth union operator.
<img alt="3d signed distance fields" src="https://user-images.githubusercontent.com/7197298/150694336-7508cb88-4262-4192-8e26-b59d2bfab436.png" width="60%">

A cloud of primitives smooth union-ed. Also shows off the layering system the renderer is capable of. You can freely combine 2D and 3D layers.
<img alt="layering of a 3D cloud scene with a 2D frame and a yin-yang symbol." src="https://user-images.githubusercontent.com/7197298/150694549-c2e37989-fff4-4629-ac03-930137774a13.png" width="60%">


Some basic UI-Elements like buttons, scroll boxes and ttf-glyphs. The bigger gray field is supposed to be a Bézier spline editor at some point.
<img alt="Simple 2D UI elements" src="https://user-images.githubusercontent.com/7197298/150694703-4a9758bd-6dc7-42a3-a0bb-a96bd80f37f6.png" width="60%">

**Comment by hatoo** [comment]
Hi, I'm developing a raytracing renderer entirely written in Rust using rust-gpu.
https://github.com/hatoo/rene

It is based on [pbrt-v3](https://github.com/mmp/pbrt-v3) and the development is still very early stage but I've succeeded to render some of the scenes from https://benedikt-bitterli.me/resources/

<img alt="cornell-box" src="https://user-images.githubusercontent.com/1667746/151118910-45d0e0ad-582c-4736-a3c1-694eefb797f3.png" width="60%">
<img alt="dragon" src="https://user-images.githubusercontent.com/1667746/151118922-3e2eefef-b20d-4771-985b-4ebde9b81e2e.png" width="60%">
<img alt="teapot" src="https://user-images.githubusercontent.com/1667746/151118926-abbf230c-5bbe-4e85-bab5-f77b6dfaf75d.png" width="60%">
